tests/test_enterprise/views/test_grant_data_sharing_permissions.py

```diff
@@ -594,13 +596,10 @@ def test_post_course_specific_consent(
 
         assert resp.url.endswith(expected_redirect_url)  # pylint: disable=no-member
         assert resp.status_code == 302
-        if not defer_creation:
-            dsc.refresh_from_db()
-            assert dsc.granted is consent_provided
```

Comment:
Can you explain why this was removed?

Reply:
I moved it to be under line 602, we don't want dsc.granted to be consent_provided because of previous conversation, but we do want to verify that it's set to True after the learner enrolls in the course.

Reply:
Yeah, that's not quite the same thing though. These lines tested whether the DSC record has been updated with `consent_provided` (given `defer_creation` is `False`), and ~~isn't~~ shouldn't be specific to `license_uuid` being passed. 

While digging to understand this I realized something potentially big - [these lines](https://github.com/edx/edx-enterprise/blob/31183667f211582ffe872740c5bde141b40e720a/enterprise/views.py#L639-L659) shouldn't be placed within the `if license_uuid:` block. See indentation from original code in [this commit diff](https://github.com/edx/edx-enterprise/commit/bf838e982476f615072fec8239ca2fc24e808463). If my assumptions are correct, enterprise course enrollments aren't possible without a `license_uuid` right now.

Reply:
Those lines aren't placed within the `if license_uuid:` block, that block is inside of `self._enroll_leaner_in_course(` which the update to the DSC record is outside of. Updating the DSC record is in line with this if statement though `if defer_creation is None and consent_record.consent_required() and consent_provided:` which is what it was originally in line with.

You had told me to put updating the granted value inside of the try block because it didn't actually matter in the original functionality failing to enroll the learner in a course would return a redirect to the failure url before it got updated anyway.

Reply:
You don't seem to be talking about the lines I linked so take a look at this picture. Highlighted lines shouldn't be at that indentation level (within the `if license_uuid:` block). 

<img width="1033" alt="Screen Shot 2020-10-22 at 2 43 39 PM" src="https://user-images.githubusercontent.com/58956521/96916218-41a51880-1475-11eb-8250-87972ac62d86.png">

Reply:
After talking to Brittney about this it isn't an urgent thing since, as far as we can tell, all enterprise enrollments are either with a `license_uuid` or `defer_creation`. Meaning, highlighted code doesn't even get hit anymore 😂 . Either way this indentation should be fixed to be consistent with previous implementation, we can look into cleaning DSC up later if Brittney and Joe so choose. 

Also, the function name `_enroll_leaner_in_course` could use some improvement - "learner" should be spelled right and since it requires `license_uuid` I propose it changes to `_enroll_licensed_learner_in_course`

Reply:
Ahhh now I see what you mean. And oops on the typo

Reply:
Well I unindented the thing, fixed the typo, changed the tests to verify that the enterprise course enrollment is created regardless of there being a license_uuid or not.